rust/scripts/bench.sh

```diff
@@ -1,44 +1,48 @@
 #!/bin/bash
 
-# Run Go interpreter benchmarks for evmrs, evmzero, lfvm and geth in branch origin/main and evmrs in branch origin/evmrs-main.
+# Run Go interpreter benchmarks for evmzero, lfvm and geth in as well as for evmrs with different features in branch origin/evmrs-main.
 # This script must be called from the parent directory of Tosca.
 
-mkdir -p benches
-
-cd Tosca
-git fetch --all
-
 RUNS=20
 TIMEOUT=1h
-REF_BRANCH="origin/main"
-NEW_BRANCH="origin/evmrs-main"
+BRANCH="origin/evmrs-main"
+
+DATE=$(date +'%Y-%m-%dT%H:%M')
 
-BRANCH=$REF_BRANCH
+BENCH_DIR=benches/$DATE
+mkdir -p $BENCH_DIR
+
+cd Tosca
+
+git fetch --all
 git checkout ${BRANCH}
-make clean-rust
+GIT_REF=$(git show-ref --hash=7 ${BRANCH})
+
 make
 
-GIT_REF=$(git show-ref --hash=7 ${BRANCH})
-INTERPRETERS=("evmrs" "evmzero" "lfvm" "geth")
+# bench all non Rust interpreters
+INTERPRETERS=("evmzero" "lfvm" "geth")
 for INTERPRETER in "${INTERPRETERS[@]}"; do
-    BENCH_NAME=$(basename ${BRANCH})-${GIT_REF}-${INTERPRETER}-${RUNS}
+    BENCH_NAME=$(basename ${GIT_REF})-${INTERPRETER}-${RUNS}
     echo running ${BENCH_NAME}
     go test ./go/integration_test/interpreter \
-        --run none -bench ^Benchmark[a-zA-Z]+/./${INTERPRETER}$ \
-        -timeout $TIMEOUT -count $RUNS | tee ../benches/${BENCH_NAME}
-    sed -i "s/${INTERPRETER}-//g" ../benches/${BENCH_NAME}
+        --run none --bench ^Benchmark[a-zA-Z]+/./${INTERPRETER}$ \
+        --timeout $TIMEOUT --count $RUNS | tee ../${BENCH_DIR}/${BENCH_NAME}
+    sed -i "s/${INTERPRETER}-//g" ../${BENCH_DIR}/${BENCH_NAME}
 done
 
-BRANCH=$NEW_BRANCH
-git checkout ${BRANCH}
-make clean-rust
-make tosca-rust
-
-GIT_REF=$(git show-ref --hash=7 ${BRANCH})
+# bench evmrs with different features
 INTERPRETER="evmrs"
-BENCH_NAME=$(basename ${BRANCH})-${GIT_REF}-${INTERPRETER}-${RUNS}
-echo running ${BENCH_NAME}
-go test ./go/integration_test/interpreter \
-    --run none -bench ^Benchmark[a-zA-Z]+/./${INTERPRETER}$ \
-    -timeout $TIMEOUT -count $RUNS | tee ../benches/${BENCH_NAME}
-sed -i "s/${INTERPRETER}-//g" ../benches/${BENCH_NAME}
+FEATURES_LIST=("" "performance")
+for FEATURES in "${FEATURES_LIST[@]}"; do
+    cd rust
+    cargo build --lib --release --features "$FEATURES"
+    cd ..
+
+    BENCH_NAME=$(basename ${GIT_REF})-${INTERPRETER}-${FEATURES}-${RUNS}
+    echo running ${BENCH_NAME}
+    go test ./go/integration_test/interpreter \
+        --run none --bench ^Benchmark[a-zA-Z]+/./${INTERPRETER}$ \
+        --timeout $TIMEOUT --count $RUNS | tee ../${BENCH_DIR}/${BENCH_NAME}
+    sed -i "s/${INTERPRETER}-//g" ../${BENCH_DIR}/${BENCH_NAME}
```

Comment:
what are you doing here?  need to patch the path?

Reply:
you mean the `sed ...`? The benchmark names contain the name of the interpreters so in order for benchstat to compare them, they must be renamed so that they all have the same names.